examples/gym/solve_gridworld.py

```diff
@@ -0,0 +1,201 @@
+        print("REWARD:", reward.__dict__)
+        logger.info(('Game Finished -- Reward {:.3f}'.format(reward.raw_reward) +
+                      ' -- Final state {}'.format(path[-1]) +
+                      ' -- CPU time {:1.3f} (s)'.format(elapsed)))
+
+def train_model():
+    construct_problem().train_policy_model(steps_per_epoch=100,
+                                           game_count_delay=10,
+                                           verbose=2)
+
+
+def monitor():
+
+    from rlmolecule.sql.tables import RewardStore
+    problem = construct_problem()
+
+    while True:
+        best_reward = problem.session.query(RewardStore) \
+            .filter_by(run_id=problem.run_id) \
+            .order_by(RewardStore.reward.desc()).first()
+
+        num_games = len(list(problem.iter_recent_games()))
+
+        if hasattr(best_reward, "data") and "position" in best_reward.data:
+            print(f"Best Reward: {best_reward.reward:.3f} for final position "
+                  f"{best_reward.data['position']} with {num_games} games played")
+
+        time.sleep(5)
+
+
+def setup_argparser():
+    
+
+    return parser
+
+
+if __name__ == "__main__":
+
+    import argparse
+    parser = argparse.ArgumentParser(
+        description='Solve the Hallway problem (move from one side of the hallway to the other). ' +
+                'Default is to run multiple games and training using multiprocessing')
+
+    parser.add_argument('--train-policy', action="store_true", default=False,
+                        help='Train the policy model only (on GPUs)')
+    parser.add_argument('--rollout', action="store_true", default=False,
+                        help='Run the game simulations only (on CPUs)')
+    parser.add_argument('--num-workers', type=int, default=7,
+                        help='Number of multiprocessing workers when running local rollout')
+    parser.add_argument("--num-games", type=int, default=None)
+    parser.add_argument("--num-mcts-samples", type=int, default=50)
+    parser.add_argument("--use-mcts", action="store_true")
+    parser.add_argument("--log-level", default="INFO", type=str)
+
+    args = parser.parse_args()
+
+    logger.setLevel(level=getattr(logging, args.log_level.upper()))
+
+    if args.train_policy:
+        train_model()
+    elif args.rollout:
+        run_games(
+            use_mcts=args.use_mcts,
+            num_mcts_samples=args.num_mcts_samples,
+            num_games=args.num_games)
+    else:
+        assert args.num_workers >= 3  # need at least 3 workers here...
+
+        import multiprocessing
+
+        jobs = [multiprocessing.Process(target=monitor)]
+        jobs[0].start()
+        time.sleep(1)
+
+        for i in range(args.num_workers - 2):
+            jobs += [multiprocessing.Process(target=run_games)]
+
+        jobs += [multiprocessing.Process(target=train_model)]
+
+        for job in jobs[1:]:
+            job.start()
+
+        for job in jobs:
+            job.join(300)
```

Comment:
todo for me, unless some else wanted to try it: does joblib's [Parallel](https://joblib.readthedocs.io/en/latest/parallel.html) (with the loky backend) throw the same tensorflow error if we don't delay imports to construct_problem()? That might simplify some of the problem initialization

Reply:
i can take a swing at this tomorrow, either way is fine with me. i was also going to add a notebook version of the gridworld problem so we have an easily runnable, interactive, self contained example